File: yujung/프로그래머스/모음 사전.cpp
#include <string>
#include <vector>

using namespace std;
string a="AEIOU";
int cnt=0;
int answer=0;
void dfs(string w,string word){
    //cnt++;
    if(w==word){answer=cnt;}
    if(w.size()>=5){return ;}
    for(int i=0; i<5;i++){
          cnt++;
    dfs(w+a[i],word);
    }
    
}
//단어 하나 word가 매개
int solution(string word) {
    dfs("",word);
    return answer;
  //     return answer-1;
}

File: seoyeon/프로그래머스/알고리즘 고득점 Kit/완전탐색/모음사전.py
#프로그래머스 알고리즘 고득점 Kit 
#완전탐색
#모음사전

alpha_words = []
alpha = ["A","E","I","O","U"]

def dfs(depth,word):
    global alpha_words,alpha
    
    alpha_words.append(word)
    
    if len(word)==5:
        return
    
    for al in alpha:
        dfs(depth+1,word+al)

def solution(word):
    answer = 0
    
    for al in alpha:
        dfs(1,al)
    
    answer = alpha_words.index(word)+1
    return answer